azuread/resource_user.go

```diff
@@ -101,6 +101,62 @@ func resourceUser() *schema.Resource {
 					"Required for users that will be assigned licenses due to legal requirement to check for availability of services in countries. " +
 					"Examples include: `NO`, `JP`, and `GB`. Not nullable.",
 			},
+
```

Comment:
These properties should probably all be computed

Reply:
Why they should be computed as they are not actually computed, By setting them as computed, We'll lose the ability to clear the value of the attributes.

Reply:
That's true, however without the Computed flag, these properties will be cleared if they are not specified in configuration, potentialy wiping all of those fields for existing users. For example:

![Screenshot 2020-06-23 01 19 23](https://user-images.githubusercontent.com/251987/85347136-9c425e00-b4ef-11ea-9ef0-dbc388bfb813.png)

Reply:
I prefer not to mark them as computed because the ability to clear up data is essential. For users who don't want to manage some attributes with terraform, there is a workaround to use `lifecycle.ignore_changes`。

Reply:
@nauxliu - these should be computed. If a user wishes to clear them couldn't they just set it to ""?

Reply:
I've tried to pass `""`  to computed attributes and some other workaround, there is no way to clear computed data.

Reply:
@manicminer what do you think?

Reply:
@nauxliu Sorry for the delay, I've done some testing on this and spoken with the SDK team. This is an unexpected SDK behavior, possibly a regression. For now we will have to settle with partial support - where clearing the attributes requires a manual or scripted action. We can unfortunately not consider adding these as non-computed. It creates the unnecessary risk of accidentally wiping these attributes for existing users, in a database that for many organisations where federation is not in use, could be a primary source of truth for this data.

To compound things a little, we are currently readying a v1.0 release of the AzureAD provider, and won't releasing any new features in the current 0.x tree. We can of course look at adding these attributes into a post-1.0 release but this PR would need to be rebased as many files have been moved around, in addition to the other changes previously requested.